source/UberRides/Model/UpfrontFare.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  Copyright © 2016 Uber Technologies, Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+@objc(UBSDKUpfrontFare) public class UpfrontFare: NSObject, Codable {
+    /// A unique upfront fare identifier.
+    @objc public private(set) var fareID: String
+
+    /// The total upfront fare value.
+    @objc public private(set) var value: Double
+
+    /// ISO 4217 currency code.
+    @objc public private(set) var currencyCode: String
+
+    /// Formatted string of estimate in local currency.
+    @objc public private(set) var display: String
+
+    /// The upfront fare expiration time
+    @objc public private(set) var expiresAt: Date
+
+    /// The components that make up the upfront fare
+    @objc public private(set) var breakdown: [UpfrontFareComponent]
+
+    enum CodingKeys: String, CodingKey {
+        case fareID = "fare_id"
+        case value = "value"
+        case currencyCode = "currency_code"
+        case display = "display"
+        case expiresAt = "expires_at"
+        case breakdown = "breakdown"
+    }
+
+    public required init(from decoder: Decoder) throws {
```

Comment:
Not sure why in this case we need a constructor, vs other examples where we initialize inline.

We should always try to be consistent with these patterns.

Reply:
I had this in here when tweaking around, but it was easier to leave it in because otherwise I would have to add default values for class parameters.

Since that was a bit confusing for code readability, I decided to add an explicit initializer in each one of these models instead of leaving them inferred.

Reply:
Yay consistency.